src/log/file_appender.cpp

```diff
@@ -82,15 +111,18 @@ namespace fc {
                    out.flush();
                    out.close();
                }
-               remove_all(link_filename);  // on windows, you can't delete the link while the underlying file is opened for writing
-               out.open( log_filename, std::ios_base::out | std::ios_base::app );
-               create_hard_link(log_filename, link_filename);
+               if (_old_log_filename != _log_filename)
+               {
+                  _old_log_filename = _log_filename;
+                  
+                  open_log_file();
+               }
```

Comment:
The logic seems a bit strange.
* Is there a chance that the file got closed but not reopened?
* Is there a chance that the file be opened again when not closed?